pkg/importer/dataStream_test.go

```diff
@@ -7,416 +7,404 @@ import (
+		replaceQEMUOperations(qemuOperations, func() {
+			err := copy(r, out, qemu)
+			if !wantErr {
+				Expect(err).NotTo(HaveOccurred())
+			} else {
+				Expect(err).To(HaveOccurred())
+			}
+		})
+	},
+		table.Entry("successfully copy reader", stringRdr, "testoutfile", false, NewFakeQEMUOperations(nil, errors.New("Shouldn't get this"), nil), false),
+		table.Entry("successfully copy qcow2 reader", testFileRdrs, "testqcow2file", true, NewFakeQEMUOperations(nil, errors.New("Shouldn't get this"), nil), false),
+		table.Entry("expect error trying to copy invalid format", testFileRdrs, "testinvalidfile", true, NewFakeQEMUOperations(nil, nil, errors.New("invalid format")), true),
+		table.Entry("expect error trying to copy qemu process fails", testFileRdrs, "testinvalidfile", true, NewFakeQEMUOperations(errors.New("exit 1"), nil, nil), true),
+	)
+})
+
+var _ = Describe("close readers", func() {
+	type args struct {
+		readers []reader
 	}
-	f()
-}
 
-func createDataStream(ep, accKey, secKey string) *DataStream {
-	dsurl, _ := ParseEndpoint(ep)
+	rdrs1 := ioutil.NopCloser(strings.NewReader("test data for reader 1"))
+	rdrs2 := ioutil.NopCloser(strings.NewReader("test data for reader 2"))
+	rdrA := reader{4, rdrs1}
+	rdrB := reader{7, rdrs2}
 
-	ds := &DataStream{
-		url:         dsurl,
-		buf:         make([]byte, image.MaxExpectedHdrSize),
-		accessKeyID: accKey,
-		secretKey:   secKey,
-	}
+	rdrsTest := []reader{rdrA, rdrB}
 
-	ds.constructReaders()
+	It("Should successfully close readers", func() {
+		err := closeReaders(rdrsTest)
+		Expect(err).NotTo(HaveOccurred())
+	})
+})
 
-	return ds
-}
+var _ = Describe("Random file name", func() {
+	const numbyte = 8
 
-func createDataStreamBytes(testfile, ep string, defaultBuf, singlereader bool) (*DataStream, []byte, error) {
-	urlPath := filepath.Join(ep, testfile)
-	if len(ep) == 0 {
-		urlPath = getURLPath(testfile)
-	}
-	testFilePath := getTestFilePath(testfile)
-	ds := createDataStream(urlPath, "", "")
+	It("create expected random name", func() {
+		randName := make([]byte, numbyte)
+		rand.Read(randName)
+		wantString := hex.EncodeToString(randName)
 
-	// if no file supplied just return empty byte stream
-	if len(testfile) == 0 && !defaultBuf {
-		return ds, nil, nil
-	}
-	if len(testfile) == 0 && defaultBuf {
-		return ds, []byte{'T', 'E', 'S', 'T'}, nil
-	}
+		got := randTmpName("testfile.img")
+		_, fn := filepath.Split(got)
 
-	f, _ := os.Open(testFilePath)
-	defer f.Close()
-	testBytes, err := ioutil.ReadAll(f)
-	if err != nil {
-		return nil, nil, fmt.Errorf("Unable to read datastream buffer")
-	}
+		Expect(len(fn)).To(Equal(len("testfile.img") + len(wantString)))
+	})
+})
 
-	return ds, testBytes, nil
-}
+var _ = Describe("Streaming Data Conversion", func() {
 
-func getFileSize(testfile string) (int, error) {
-	f, err := os.Open(getTestFilePath(testfile))
-	defer f.Close()
+	tinyCoreIso := "tinyCore.iso"
+	baseTestImage, err := filepath.Abs(filepath.Join(TestImagesDir, tinyCoreIso))
```

Comment:
Could use `imageDir` instead of `TestImageDir` and not need to call `Abs()`

Reply:
Done